design-library/src/components/BccHeader/BccHeader.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { describe, it, expect } from "vitest";
+
+import { mount } from "@vue/test-utils";
+import BccHeader from "./BccHeader.vue";
+
+describe("BccHeader", () => {
+  it("renders title", async () => {
+    const wrapper = mount(BccHeader, { props: { title: "Title" } });
+    expect(wrapper.html()).toContain("Title</h2>");
+  });
+
+  it("renders all text", async () => {
+    const wrapper = mount(BccHeader, {
+      props: { title: "Title", overline: "Overline", underline: "Underline" },
+    });
+    expect(wrapper.html()).toContain("Overline</p>");
+    expect(wrapper.html()).toContain("Title</h2>");
+    expect(wrapper.html()).toContain("Underline</p>");
```

Comment:
It's not totally necessary to test for the actual tags - just testing if `wrapper.text()` contains the specified text is enough I think. Otherwise the test becomes dependent on the actual layout of the component which could change (or maybe we make the heading level configurable in the future for example).